src/Shared/PackageManagers/GitHubProjectManager.cs

```diff
@@ -76,14 +76,14 @@ public override async Task<IEnumerable<string>> DownloadVersion(PackageURL purl,
 
             try
             {
-                var url = $"{ENV_GITHUB_ENDPOINT}/{purl.Namespace}/{purl.Name}";
+                var url = $"{ENV_GITHUB_ENDPOINT}/{purl?.Namespace}/{purl?.Name}";
                 var invalidChars = Path.GetInvalidFileNameChars();
 
                 // TODO: Externalize this normalization
-                var fsNamespace = new String(purl.Namespace.Select(ch => invalidChars.Contains(ch) ? '_' : ch).ToArray());
-                var fsName = new String(purl.Name.Select(ch => invalidChars.Contains(ch) ? '_' : ch).ToArray());
-                var fsVersion = new String(purl.Version.Select(ch => invalidChars.Contains(ch) ? '_' : ch).ToArray());
-                var workingDirectory = string.IsNullOrWhiteSpace(purl.Version) ?
+                var fsNamespace = new string((purl?.Namespace?.Select(ch => invalidChars.Contains(ch) ? '_' : ch) ?? Array.Empty<char>()).ToArray());
```

Comment:
This can lead to a very different result if purl is null, and cause no exceptions, correct?

Reply:
If Purl is null it will set `fsNamespace` to an empty string.

Yes, this means this won't throw exceptions here if Purl is null.

Reply:
If we should throw exceptions here we need to handle them explicitly.  I can restore the old "throw an exception" behavior if that's desired.

Reply:
yes, I think we need to do so...if we have a logic where it shouldn't proceed without a valid value, we should call object.hasValue() (??) and return failure/throw exception

Reply:
This is fixed now. Check the latest.